### PR TITLE
Design improvements

### LINKED ISSUE #1: Design ideas
Hello :wave: 

Here are some design ideas for the UI:

- use a green pill instead of an orange banner
- use a rounded expanded list or an invsible expanded list
- use more yaru icons
- make the appbar less fat
- add more yaru icons to the map

Tinkered in my fork a bit for results:

- No background (my fav):
grafik
- Border radius only for first and last tile (also nice but I am not sure):
grafik

CC @jpnurmi @kenvandine

### PR DESCRIPTION
**Pull request by Feichtmeier**
- use more yaru icons
- update yaru icons dep
- use [yaru_widgets.dart](https://github.com/ubuntu/yaru_widgets.dart)
- use YaruExpansionPanelList
- reduce AppBar height
- let SmallChips have a optional color
- use a green small chip for updateable rows
- reduce padding

| Before | After |
|----|----|
| grafik | grafik |
| Bildschirmfoto von 2022-01-30 11-32-35 | Bildschirmfoto von 2022-01-30 11-53-13 |

Closes #1

### CODE REVIEW

File: lib/page.dart [hunk 1]
```diff
@@ -46,6 +47,7 @@ class _FwupdPageState extends State<FwupdPage> {
     final model = context.watch<FwupdModel>();
     return Scaffold(
       appBar: AppBar(
+        toolbarHeight: 48,
```
Comment (jpnurmi) [inline lib/page.dart:50]: is this something that could be done in yaru?

File: lib/src/widgets/device_header.dart [hunk 2]
```diff
@@ -22,18 +23,22 @@ class DeviceHeader extends StatelessWidget {
...
+              alignment: Alignment.centerRight,
+              child: SmallChip(
+                text: 'Update available',
+                color: Theme.of(context)
+                        .elevatedButtonTheme
+                        .style!
+                        .backgroundColor!
+                        .resolve({MaterialState.focused}) ??
+                    Theme.of(context).primaryColor,
+              )),
```
Comment (jpnurmi) [inline lib/src/widgets/device_header.dart:41]: could you add a comma between the parentheses to fix the indentation, please?

File: lib/page.dart (hunk 1 above)
Comment (Feichtmeier) [reply lib/page.dart:50]: I'll try :) thanks

File: lib/src/widgets/device_icon.dart [hunk 3]
```diff
@@ -7,6 +7,8 @@ const yaruIcons = <String, IconData>{
   'drive-harddisk': YaruIcons.drive_harddisk,
   'input-gaming': YaruIcons.input_gaming,
   'preferences-desktop-keyboard': YaruIcons.input_keyboard,
+  'touchpad-disabled': YaruIcons.fingerprint,
+  'thunderbolt': YaruIcons.thunderbolt
```
Comment (jpnurmi) [inline lib/src/widgets/device_icon.dart:11]: nice :+1: there's a lot more if you have time and interest :smile: 

https://lvfs.readthedocs.io/en/latest/metainfo.html#device-icons

Comment (jpnurmi) [reply lib/src/widgets/device_icon.dart]: woah, thanks!

File: lib/page.dart (hunk 1 above)
Comment (Feichtmeier) [reply lib/page.dart]: done! :)

**Comment by Feichtmeier** [comment]
ok wait need to fix the light theme

**Comment by Feichtmeier** [comment]
Updated before/after pictures with the light theme version

**Comment by Feichtmeier** [comment]
Note: there is something wrong with the finger print reader icon - the type says "touchpad-disabled" even if it should be "fingerprint-reader". Both entries have the correct yaru icon now, but in the app you get the fingerprint reader label and the touchpad icon (because it is mapped like this) :thinking: 
So either the type is wrong or the label

**Comment by jpnurmi** [comment]
Thanks for taking the time to improve the original engineer-designed looks by yours truly! :smile: 

I like the new cleaner looks but there's one significant behavioral change. Before there was a visual clue separating the devices _and_ it was possible to click anywhere on a device tile, including the name of the device. This made it easy to select a specific device. Now, it's only possible to click the small expansion icon far on the right. There's no visual clue between the devices, so it makes you glance back and forth to match the vertical level of a device and its expansion icon. Is it possible to restore the old behavior to allow clicking the devices so that the expansion icon is more like a clue that they can be expanded, not the only means to expand a device? Perhaps some subtle visual clue would also help a bit to make it easier to see what is what? Subtle hover color, separator, or something like that?

**Comment by Feichtmeier** [comment]
oh indeed - I'll fix the click area

I don't think a separator is really needed though, I'll see what can be done

**Comment by jpnurmi** [comment]
What about something like a subtle circle around the respective expansion icon when hovering a tile to indicate that the tile is interactive? :)

**Comment by Feichtmeier** [comment]
nice idea!

Did you said you wanted a separation between the rows?
Do you like a little space?

grafik

**Comment by jpnurmi** [comment]
I can see now that your first proposal looks prettier. :wind_face: Perhaps making the whole tile interactive and adding a small hover effect for the arrow would be sufficient?

**Comment by Feichtmeier** [comment]
ah okay, so you want that users understand where a tile ends and where it starts when they hover only?

**Comment by jpnurmi** [comment]
For example, how about something like this?

image

**Comment by jpnurmi** [comment]
It's the same hover effect made with GIMP. :smile:

**Comment by Feichtmeier** [comment]
I would go even further, since you are def. right that the visual hint is missing:

grafik

is this also okay for you?

**Comment by jpnurmi** [comment]
Looks good! :+1:

**Comment by Feichtmeier** [comment]
Cool! 
since it was a change in yaru_widgets.dart you can merge if you like :t-rex: 

Here is the dark version
grafik

File: lib/src/widgets/device_header.dart [hunk 4]
```diff
@@ -22,16 +23,21 @@ class DeviceHeader extends StatelessWidget {
           title: Text(device.name),
           subtitle: Text(device.summary ?? ''),
           leading: DeviceIcon.fromName(device.icon.firstOrNull),
-          contentPadding: const EdgeInsets.only(left: 24),
+          contentPadding: const EdgeInsets.only(
+            left: 24,
```
Comment (jpnurmi) [inline lib/src/widgets/device_header.dart:27]: nit-pick: looks like an unrelated change :)

File: pubspec.yaml [hunk 5]
```diff
@@ -17,8 +17,11 @@ dependencies:
       ref: wip
   path: ^1.8.0
   provider: ^6.0.0
-  yaru: ^0.2.0
-  yaru_icons: ^0.1.0
+  yaru:
+    git: http://github.com/ubuntu/yaru.dart
```
Comment (jpnurmi) [inline pubspec.yaml:21]: will you publish a yaru update?

**Review by jpnurmi — COMMENTED** [review]
Ok, thanks. LGTM :+1: Will you publish a yaru update?

File: pubspec.yaml (hunk 5 above)
Comment (Feichtmeier) [reply pubspec.yaml]: done

**Comment by jpnurmi** [comment]
Awesome! Thank you! :rocket:
